File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    var result = -1

    var count int
    for i := 0; i < len(haystack); i++ {
        count = 0
        for j := 0; j < len(needle); j++ {
            if i+j >= len(haystack) {
                return result
            } else if string(haystack[i + j]) == string(needle[j]){
                count++
            }
            if count == len(needle) {
                result = i
                break
            }
        }
        if result != -1 {
            break
        }
    }

    return result
}

File: LeetCode/Easy/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        let mut result = -1;

        for i in 0..haystack.len() {
            let mut count = 0;
            for (j, c) in needle.chars().enumerate() {
                if i + j >= haystack.len() {
                    return result;
                } else if haystack.chars().nth(i + j).unwrap() == c {
                    count += 1;
                }
                if count == needle.len() {
                    result = i as i32;
                    break;
                }
            }
            if result != -1 {
                break;
            }
        }

        result
    }
}

